lib/Connector/XiboAudienceReportingConnector.php

```diff
@@ -64,4 +111,195 @@ public function processSettingsForm(SanitizerInterface $params, array $settings)
         // TODO: Implement processSettingsForm() method.
         return $settings;
     }
+
+    /**
+     * @throws NotFoundException
+     * @throws GeneralException
+     */
+    public function onRegularMaintenance(MaintenanceRegularEvent $event)
+    {
```

Comment:
We should only do this if the connector is enabled and if we have an API key

Reply:
Same for this, how can we validate whether the connector is enabled or not

Reply:
The connector has an enabled flag, and you can test if the apiKey is empty or not.